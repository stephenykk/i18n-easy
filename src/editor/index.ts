import { flatten } from 'lodash'
import annotation from './annotation'
import completion from './completion'
import extract from './extract'
import definition from './definition'
import refactor from './refactor'
import problems from './problems'
import reference from './reference'
import statusbar from './statusbar'
import reviewComments from './reviewComments'
import { ExtensionModule } from '~/modules'

const m: ExtensionModule = (ctx) => {
  return flatten([
    annotation(ctx),
    completion(ctx),
    extract(ctx),
    refactor(ctx),
    definition(ctx),
    problems(ctx),
    reference(ctx),
    statusbar(ctx),
    reviewComments(ctx),
  ])
}

export default m
