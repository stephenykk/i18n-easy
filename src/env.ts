// @ts-ignore
export const isProd: boolean = process.env.NODE_ENV === 'production'
export const isDev = !isProd
// @ts-ignore
export const isTest: boolean = process.env.NODE_ENV === 'test'
