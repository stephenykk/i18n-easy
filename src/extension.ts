import { ExtensionContext } from 'vscode'
import { flatten } from 'lodash'
import { version } from '../package.json'
import { Global, Config, KeyDetector, CurrentFile } from '~/core'
import commandsModules, { Commands } from '~/commands'
import viewsModules from '~/views'
import { Log } from '~/utils'
import i18n from '~/i18n'
import editorModules from '~/editor'

export async function activate(ctx: ExtensionContext) {
  Log.info(`ðŸˆ¶ Activated, v${version}`)

  Config.ctx = ctx

  i18n.init(ctx.extensionPath)
  KeyDetector.init(ctx)

  // activate the extension
  await Global.init(ctx)
  CurrentFile.watch(ctx)

  const modules = [
    commandsModules,
    editorModules,
    viewsModules,
  ]

  const disposables = flatten(modules.map(m => m(ctx)))
  disposables.forEach(d => ctx.subscriptions.push(d))
}

export function deactivate() {
  Log.info('ðŸˆš Deactivated')
}

export {
  Global,
  CurrentFile,
  KeyDetector,
  Config,
  Log,
  Commands,
}
