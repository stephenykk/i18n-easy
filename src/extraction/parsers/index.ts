import * as html from './html'
import * as babel from './babel'

export const extractionsParsers = {
  html,
  babel,
}
