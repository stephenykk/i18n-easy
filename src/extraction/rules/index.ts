import { BasicExtrationRule } from './basic'
import { NonAsciiExtractionRule } from './non-ascii-characters'
import { DynamicExtractionRule } from './dynamic'

export * from './base'
export * from './basic'
export * from './non-ascii-characters'

export const DefaultExtractionRules = [
  new BasicExtrationRule(),
  new NonAsciiExtractionRule(),
]

export const DefaultDynamicExtractionsRules = [
  new DynamicExtractionRule(),
]
