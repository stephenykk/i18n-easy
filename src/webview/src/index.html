<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf8" />
    <meta name="viewport" content="width=device-width" />
  </head>

  <body>
    <div id="app"></div>
    <script src="./index.js"></script>
  </body>
</html>
