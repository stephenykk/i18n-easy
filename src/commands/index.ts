import { flatten } from 'lodash'
import configLocales from './configLocalePaths'
import configLanguages from './configLanguages'
import keyManipulations from './keyManipulations'
import extractText from './extractString'
import detectHardStrings from './detectHardStrings'
import help from './help'
import refreshUsageReport from './refreshUsageReport'
import editor from './openEditor'
import review from './review'
import deepl from './deepl'
import gotoRange from './gotoRange'
import gotoNextUsage from './gotoNextUsage'
import batchHardStringsExtract from './extractStringBulk'
import { ExtensionModule } from '~/modules'

const m: ExtensionModule = (ctx) => {
  return flatten([
    configLocales(ctx),
    configLanguages(ctx),
    keyManipulations(ctx),
    extractText(ctx),
    detectHardStrings(ctx),
    batchHardStringsExtract(ctx),
    help(ctx),
    refreshUsageReport(ctx),
    editor(ctx),
    review(ctx),
    deepl(ctx),
    gotoRange(ctx),
    gotoNextUsage(ctx),
  ])
}

export * from './commands'

export default m
