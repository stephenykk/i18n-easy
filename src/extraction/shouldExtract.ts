import { DefaultExtractionRules, ExtractionRule, ExtractionScore } from './rules'

export function shouldExtract(str: string, rules: ExtractionRule[] = DefaultExtractionRules): boolean {
  let extract = false
  for (const rule of rules) {
    const result = rule.shouldExtract(str)
    if (result === ExtractionScore.MustExclude)
      return false
    if (result === ExtractionScore.MustInclude)
      return true
    if (result === ExtractionScore.ShouldExclude)
      extract = false
    if (result === ExtractionScore.ShouldInclude)
      extract = true
  }
  return extract
}
