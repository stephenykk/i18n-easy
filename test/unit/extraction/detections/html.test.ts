import { expect } from 'chai'
import { extractionsParsers } from '../../../../src/extraction'

const html = extractionsParsers.html

describe('detections', () => {
  describe('attrs', () => {
    it('html', () => {
      expect(html.detect('<div placeholder="Hello"></div>')).to.matchSnapshot()

      expect(html.detect(`
<textarea
  class="form-control"
  rows="8"
  v-model="article.body"
  placeholder="Write your article (in markdown)"
  ></textarea>
`)).to.matchSnapshot()

      expect(html.detect('<img src="{{ img_url }}" title="This is a title" alt="This is an alt.">')).to.matchSnapshot()
      expect(html.detect('<img src="https://example.com/img.svg" :title="image" alt="img.svg">')).eql([])
    })

    it('vue-favored attrs', () => {
      expect(html.detect(
        // eslint-disable-next-line no-template-curly-in-string
        '<input v-bind:placeholder="\'Placeholder \' + input_name" :title="`${input_name} title`" name="email">'),
      ).to.matchSnapshot()
    })
  })

  describe('inline', () => {
    it('plain text', () => {
      expect(html.detect('<p>Popular Tags</p>')).to.matchSnapshot()
    })

    it('mixed text', () => {
      expect(html.detect(`
<p>
  <router-link :to="{ name: 'login' }">Sign in</router-link>
  or
  <router-link :to="{ name: 'register' }">sign up</router-link>
  to add comments on this article.
</p>`)).to.matchSnapshot()
    })

    it('vue-favored variables', () => {
      expect(html.detect('<p>Using mustaches: {{ rawHtml }}</p>')).to.matchSnapshot()
    })

    it('ignore translated keys', () => {
      expect(html.detect('<p>{{ $t("hello.world") }}</p>').length).to.eq(0)
      expect(html.detect('<p :placeholder="t(`hello.world`)"></p>').length).to.eq(0)
      expect(html.detect('<p :placeholder="hello + world"></p>').length).to.eq(0)
    })

    it('multiple lines', () => {
      expect(html.detect(`
<p>
  Using
  v-html
  directive:
  <span v-html="rawHtml"></span>
</p>
`)).to.matchSnapshot()
    })

    it('invalid', () => {
      expect(html.detect('<p>class.name</p>')).eql([])
      expect(html.detect('<p> {{ $t("hello") }} </p>')).eql([])
    })

    it('exclude tag', () => {
      expect(html.detect('<script>Hello</script>')).eql([])
      expect(html.detect('<style lang="ts">Hello</style>')).eql([])
    })
  })
})
