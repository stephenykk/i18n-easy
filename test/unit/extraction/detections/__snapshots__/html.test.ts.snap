// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detections attrs html 1`] = `
Array [
  Object {
    "end": 23,
    "fullEnd": 24,
    "fullStart": 5,
    "fullText": "placeholder=\\"Hello\\"",
    "isDynamic": false,
    "source": "html-attribute",
    "start": 18,
    "text": "Hello",
  },
]
`;

exports[`detections attrs html 2`] = `
Array [
  Object {
    "end": 117,
    "fullEnd": 118,
    "fullStart": 72,
    "fullText": "placeholder=\\"Write your article (in markdown)\\"",
    "isDynamic": false,
    "source": "html-attribute",
    "start": 85,
    "text": "Write your article (in markdown)",
  },
]
`;

exports[`detections attrs html 3`] = `
Array [
  Object {
    "end": 47,
    "fullEnd": 48,
    "fullStart": 25,
    "fullText": "title=\\"This is a title\\"",
    "isDynamic": false,
    "source": "html-attribute",
    "start": 32,
    "text": "This is a title",
  },
  Object {
    "end": 69,
    "fullEnd": 70,
    "fullStart": 49,
    "fullText": "alt=\\"This is an alt.\\"",
    "isDynamic": false,
    "source": "html-attribute",
    "start": 54,
    "text": "This is an alt.",
  },
]
`;

exports[`detections attrs vue-favored attrs 1`] = `
Array [
  Object {
    "end": 54,
    "fullEnd": 55,
    "fullStart": 7,
    "fullText": "v-bind:placeholder=\\"'Placeholder ' + input_name\\"",
    "isDynamic": true,
    "source": "html-attribute",
    "start": 27,
    "text": "'Placeholder ' + input_name",
  },
  Object {
    "end": 85,
    "fullEnd": 86,
    "fullStart": 56,
    "fullText": ":title=\\"\`\${input_name} title\`\\"",
    "isDynamic": true,
    "source": "html-attribute",
    "start": 64,
    "text": "\`\${input_name} title\`",
  },
]
`;

exports[`detections inline mixed text 1`] = `
Array [
  Object {
    "end": 51,
    "fullText": "Sign in",
    "source": "html-inline",
    "start": 44,
    "text": "Sign in",
  },
  Object {
    "end": 120,
    "fullText": "sign up",
    "source": "html-inline",
    "start": 113,
    "text": "sign up",
  },
  Object {
    "end": 170,
    "fullText": "
  to add comments on this article.
",
    "source": "html-inline",
    "start": 134,
    "text": "
  to add comments on this article.
",
  },
]
`;

exports[`detections inline multiple lines 1`] = `
Array [
  Object {
    "end": 37,
    "fullText": "
  Using
  v-html
  directive:
  ",
    "source": "html-inline",
    "start": 4,
    "text": "
  Using
  v-html
  directive:
  ",
  },
]
`;

exports[`detections inline plain text 1`] = `
Array [
  Object {
    "end": 15,
    "fullText": "Popular Tags",
    "source": "html-inline",
    "start": 3,
    "text": "Popular Tags",
  },
]
`;

exports[`detections inline vue-favored variables 1`] = `
Array [
  Object {
    "end": 33,
    "fullText": "Using mustaches: {{ rawHtml }}",
    "source": "html-inline",
    "start": 3,
    "text": "Using mustaches: {{ rawHtml }}",
  },
]
`;
